src/mongocxx/instance.cpp

```diff
@@ -90,7 +91,15 @@ class instance::impl {
         } else {
             libmongoc::log_set_handler(null_log_handler, nullptr);
         }
-        libmongoc::handshake_data_append("mongocxx", MONGOCXX_VERSION_STRING, NULL);
+
+        // Despite the name, mongoc_handshake_data_append *prepends* the platform string.
+        // mongoc_handshake_data_append does not add a delimitter, so include the " / " in the
+        // argument for consistency with the driver_name, and driver_version.
+        std::stringstream platform;
+        platform << "CXX=" << MONGOCXX_COMPILER_ID << " " << MONGOCXX_COMPILER_VERSION << " "
+                 << "stdcxx=" << __cplusplus << " / ";
```

Comment:
Possible tweak: The `__cplusplus` macro is a lie for MSVC (by default). It would be useful to add a field for `_MSVC_LANG`, which tells the true value of `/std:` regardless of the compiler conformance options.

Reply:
Good point. The installation documentation suggests compiling with `/Zc:__cplusplus` for C++17. But users may be building with older versions.

Updated to prefer `_MSVC_LANG`. Filed CXX-2629 to consider setting `/Zc:__cplusplus` as the default when building with MSVC.